core/src/test/java/cucumber/runner/TimeServiceStub.java

```diff
@@ -0,0 +1,18 @@
+package cucumber.runner;
+
+public class TimeServiceStub implements TimeService {
+    private final long duration;
+    private final ThreadLocal<Long> currentTime = new ThreadLocal<Long>();
```

Comment:
The amount of ThreadLocals I'm seeing makes me itch. We'll have to look at that time time in the future.